src/Umbraco.Core/Composing/CompositionExtensions.cs

```diff
@@ -10,32 +11,32 @@ public static class CompositionExtensions
         /// <summary>
         /// Sets the published snapshot service.
         /// </summary>
-        /// <param name="composition">The composition.</param>
+        /// <param name="builder">The builder.</param>
         /// <param name="factory">A function creating a published snapshot service.</param>
-        public static void SetPublishedSnapshotService(this Composition composition, Func<IServiceProvider, IPublishedSnapshotService> factory)
+        public static void SetPublishedSnapshotService(this IUmbracoBuilder builder, Func<IServiceProvider, IPublishedSnapshotService> factory)
```

Comment:
I know you didn't write this but I'm always concerned when I see `Func<IServiceProvider, TResult>` used for service registration. It limits the scope of the result to Singleton.

Reply:
Noted, but out of scope for this PR :)